source/common/http/http1/codec_impl.cc

```diff
@@ -164,6 +164,10 @@ void StreamEncoderImpl::encodeData(Buffer::Instance& data, bool end_stream) {
 
 void StreamEncoderImpl::encodeTrailers(const HeaderMap&) { endEncode(); }
 
+void StreamEncoderImpl::encodeMetadata(const MetadataMapVector&) {
+  ENVOY_LOG_MISC(error, "HTTP1 doesn't support metadata");
```

Comment:
So if an H2 upstream sends medatadata to an H1 data source, we'll log an error message for every frame?  I think we want to handle that in a way that's less painful.  I think ideally we'd error log once (or better yet with pow-2 backoff) and increment a stat, but I don't think Envoy has backoff logging yet so maybe just a stat for now?

Reply:
Good idea! Fixed in a separate [PR](https://github.com/envoyproxy/envoy/pull/7801).

Reply:
Cool.  If you want to change this to a no-op I'd be fine with it landing as-is, though it should probably get a Matt Klein pass as well.

Reply:
Maybe I can sync to master after #7801 is merged?

Reply:
Yeah agreed this should be a NOP + stat, so fine to review the other change first and then come back to this one.